addons/mail/static/src/model/model_field.js

```diff
@@ -792,97 +656,81 @@ class ModelField {
      *
      * @private
      * @param {mail.model} record
-     * @param {string|mail.model|<string|mail.model>[]|null} newValue
+     * @param {mail.model|mail.model[]|null} newValue
      */
     _setRelationUnlinkX2Many(record, newValue) {
-        // convert newValue to array of localId, null is considered unlink all
-        const otherLocalIds = newValue === null
-            ? [...this.read(record)]
+        // null is considered unlink all
+        const deleteOtherRecords = newValue === null
+            ? this.read(record)
```

Comment:
We no longer want to support `null` value. I think it's easy to remove, because all remaining usage and/or support were in this file when I checked a few days ago.